apps/dashr-ocean-optics/app.R

```diff
@@ -0,0 +1,574 @@
+library(dash)
+library(dashDaq)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+
+if (!appName == "") {
+  pathPrefix <- sprintf("/%s/", appName)
+
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+# colors for layout and figure
+colorsRaw <- read.table("colors.txt", sep = ' ', comment.char = '',
```

Comment:
Not an important comment, but just out of curiosity, is there a reason to store the color data in a separate file? Why not just use a list like:

```
colors <- list(
  background = "#bbbbbb",
  primary = "#ffffff",
  secondary = "#efefef",
  tertiary = "#dfdfdf",
  grid-colour = "#eeeeee",
  accent = "#2222ff"
)
```

Reply:
Since Python version was using a separate `colors.txt`, I did it the same way. Also, I wouldn't mind defining it as a `list` in app.R :)